pkg/cmd/server/origin/run_components.go

```diff
@@ -167,7 +168,12 @@ func (c *MasterConfig) RunDNSServer() {
 	}
 
 	go func() {
-		err := dns.ListenAndServe(config, c.DNSServerClient(), c.EtcdClient)
+		etcdClient, err := etcd.GetAndTestEtcdClient(c.Options.EtcdClientInfo)
```

Comment:
It's so that if necessary an admin could do a customization to have SkyDNS
serve some additional routes.  Not used today, let's open a follow up to
track it.

On Fri, Feb 12, 2016 at 11:13 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/server/origin/run_components.go
> <https://github.com/openshift/origin/pull/6933#discussion_r52760912>:
>
> > @@ -167,7 +168,12 @@ func (c *MasterConfig) RunDNSServer() {
> >  	}
> >
> >  	go func() {
> > -		err := dns.ListenAndServe(config, c.DNSServerClient(), c.EtcdClient)
> > +		etcdClient, err := etcd.GetAndTestEtcdClient(c.Options.EtcdClientInfo)
>
> I'm actually wondering why we do this at all... @smarterclayton
> <https://github.com/smarterclayton>, is there a reason we're not giving
> skynet a nil etcdclient?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6933/files#r52760912>.
>